polyfill/intersection-observer.js

```diff
@@ -965,16 +965,16 @@ function getParentNode(node) {
     return getFrameElement(node);
   }
 
+  // If the parent has element that is assigned through shadow root slot
+  if (parent && parent.assignedSlot) {
+    parent = parent.assignedSlot.parentNode
```

Comment:
I don't think this is correct if the assigned slot has a box (via e.g. `slot { display: block }`).

Reply:
> I don't think this is correct if the assigned slot has a box (via e.g. slot { display: block }).

Not sure if anyone style slot element